enhancements/olm/operator-must-gather.md

```diff
@@ -0,0 +1,137 @@
+---
+title: operator-must-gather
+authors:
+  - "@shawn-hurley"
+reviewers:
+  - "@ecordell"
+  - "@njhale"
+approvers:
+  - "@ecordell"
+  - "@njhale"
+creation-date: 2020-1-6
+last-updated: 2020-1-6
+status: implementable
+see-also:
+  - "./enhancements/must-gather.md"
+  - "./enhancements/oc/inspect.md"
+---
+
+# Operator Must Gather
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+## Summary
+
+Components developing [must gather images]("../oc/must-gather.md#must-gather-images") to gather all the information that they will need for debugging has become the defacto mechanism for customers to father information for a support case. When debugging installation of an operator with OLM we need to gather data from objects across the cluster. 
+
+## Motivation
+
+Today back and forth between the customer and the operator team or the OLM team to get info about an operator failing install. This the current state or each operator needs to write their own must gather to capture everything. Operator authors should have a starting point for writing their own must gathers such that gathering the OLM information for their operator should be consistent. 
+
+
+### Goals
+
+List the specific goals of the proposal. How will we know that this has succeeded?
+1. Gathering should gather exactly the operator, and relatedObjects for that operator, and the resources created by OLM.
+2. A library that can be used to enable this functionality across multiple commands.
+3. A small set of user facing CLI flags. 
+4. Integaration in a seperate must-gather image for other operator authors, or via the library should
+5. Should be 100% functional when not in a openshift environment.
+
+### Non-Goals
+
+1. Anything related to the wider cluster. This is not meant to be a OCP debug tool.
+2. Specficics for a given operator. This is meant to help debug install of the operator, not the operator's functionality. 
+3. Handle low level install scenarios, such as manually creating a CSV after applying RBAC.
+
+## Proposal
+
+A library, in the [api](https://github.com/operator-framework/api/tree/master/pkg) repo that will be vendored by `OLM` and `operator-sdk`. This library will include all of the traversal logic for retrieving OLM Operators data. 
```

Comment:
I'm not 100% sure this is the right place, but my opinion will be based on what changes would be required in `https://github.com/operator-framework/api/blob/master/go.mod`

(we want the api types to be importable in as many places as possible)

Reply:
that sounds like a plan if we end up bringing in to many dependencies then we can re-evaluate the location.